source/flux/parts/buildings.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+
+import * as storage from '../../lib/storage'
+import type {BuildingType} from '../../views/building-hours/types'
+
+const SAVE_FAVORITE_BUILDINGS = 'buildings/SAVE_FAVORITE_BUILDINGS'
+
+export async function loadFavoriteBuildings() {
+  let favoriteBuildings = await storage.getFavoriteBuildings()
+  return saveFavoriteBuildings(favoriteBuildings)
+}
+
+type SaveFavoriteBuildingsAction = {
+  type: 'buildings/SAVE_FAVORITE_BUILDINGS',
+  payload: Array<BuildingType>,
```

Comment:
It's probably significantly simpler overall to store an array of the building names as strings instead of the actual building information. Also requires less space on the device and doesn't raise so many issues of caching. 

Would you mind working this over to use an array of strings instead of BuildingType, please?

Reply:
Well, so much for the "start a review" button… Review in progress!